### ISSUE TITLE
Add-on does not work and lock "Preferences" in Firefox anymore from v.37 & up

### ISSUE THREAD

**Issue by ayrtonsf1** [mrigor/publicfox#21]
Firefox has redesigned its toolbar and it has a new UI for "Preferences". publicfox no longer password protects and locks up preferences for Firefox version 37 & up.

**Comment by windblife** [comment]
Yes, on Firefox 38, when I go Options, it opens a new preferences tab (about:preferences) instead of asking for publicfox password.
Please fix ASAP and also make it compatible with older version as it is now.
Would be nice if you can also block About Firefox menu, to prevent Firefox update.
Thanks a lot!

**Comment by ggilbert1** [comment]
Yes please fix this! Have Firefox on some public computers and need to keep users from seeing the password for a paid database. Also from messing around with things in general.

**Comment by alan8wolker** [comment]
As a temporary measure:
to solve this problem restoring the classic start menu settings
in about:config, find "browser.preferences.inContent" and set the value to "false"
after restarting firefox addon starts to work correctly.
tested on  firefox version 40.0.2

**Comment by deltozavr** [comment]
in dlwatchoverlay.js add in dlwatch['checkurl'] = function()
after line 160
(location.toLowerCase().indexOf("about:support") != -1 &amp;&amp; dlwatchPref.getBoolPref("addonslock")) ||
(location.toLowerCase().indexOf("about:preferences") != -1 &amp;&amp; dlwatchPref.getBoolPref("addonslock")) ||
(location.toLowerCase().indexOf("about:accounts") != -1 &amp;&amp; dlwatchPref.getBoolPref("addonslock")) ||
it will lock about:preferences, about:support, about:accounts !!!

**Comment by ggilbert1** [comment]
Thank you alan8wolker, worked perfectly, just what was needed.

**Comment by mangstadt** [comment]
Hello,

Is there any update on this issue?  I work in a public access computer lab and we have been relying on Public Fox to lock down our Firefox installations.  Thank you!

We are currently running version 41.0.2 of Firefox, and the "browser.preferences.inContent" setting @alan8wolker mentioned works perfectly!  However, that setting does not exist in the latest version of Firefox (43.0.3 at the time of this writing).
